org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/internal/workspace/SynchronizeGradleBuildOperation.java

```diff
@@ -240,11 +249,74 @@ private IProject addNewEclipseProjectToWorkspace(EclipseProject project, SubMoni
     }
 
     private void uncoupleWorkspaceProjectFromGradle(IProject workspaceProject, SubMonitor monitor) {
-        monitor.setWorkRemaining(3);
+        monitor.setWorkRemaining(4);
         monitor.subTask(String.format("Uncouple workspace project %s from Gradle", workspaceProject.getName()));
         CorePlugin.workspaceOperations().refreshProject(workspaceProject, monitor.newChild(1, SubMonitor.SUPPRESS_ALL_LABELS));
+        unconfigureConfigurators(workspaceProject, monitor.newChild(1));
         CorePlugin.workspaceOperations().removeNature(workspaceProject, GradleProjectNature.ID, monitor.newChild(1, SubMonitor.SUPPRESS_ALL_LABELS));
         CorePlugin.modelPersistence().deleteModel(workspaceProject);
         CorePlugin.configurationManager().deleteProjectConfiguration(workspaceProject);
     }
+
+    private void initConfigurators(IProgressMonitor monitor) {
+        GradleBuild gradleBuild = GradleCore.getWorkspace().createBuild(this.buildConfig.toApiBuildConfiguration());
+        InitializationContext context = new InitializationContext() {
+
+            @Override
+            public GradleBuild getGradleBuild() {
+                return gradleBuild;
+            }
+        };
+
+        SubMonitor progress = SubMonitor.convert(monitor);
+        progress.setWorkRemaining(this.configurators.size());
+        for (ProjectConfigurator c : this.configurators) {
+            try {
+                c.init(context, progress.newChild(1));
+            } catch (Exception e) {
+                CorePlugin.logger().warn("Project configurator " + c.getClass().getName() + " failed to initialize" , e);
```

Comment:
We'll need some user-visible indication, e.g. an error marker on the project or similar.

Reply:
Isn't it enough if we log the problem? I'm not really sure if we want to use error markers for this. In the end, it's the contributors' responsibility to provide exception handling imo.

Reply:
If we get an exception here it's because the contributor didn't handle exceptions correctly. We don't want to just silently pretend that everything is alright. Users won't look in the warning log, they'll just wonder why some things don't quite work as expected.